util/src/main/java/com/google/cloud/hadoop/util/interceptors/InvocationIdInterceptor.java

```diff
@@ -0,0 +1,51 @@
+package com.google.cloud.hadoop.util.interceptors;
+
+import com.google.api.client.http.HttpExecuteInterceptor;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import java.io.IOException;
+import java.util.UUID;
+
+public final class InvocationIdInterceptor implements HttpExecuteInterceptor {
+
+  public static final String GCCL_INVOCATION_ID_PREFIX = "gccl-invocation-id/";
+
+  public static final String GOOG_API_CLIENT = "x-goog-api-client";
+
+  private final HttpExecuteInterceptor interceptor;
+
+  public InvocationIdInterceptor(HttpExecuteInterceptor interceptor) {
+    this.interceptor = interceptor;
+  }
+
+  @Override
+  public void intercept(HttpRequest request) throws IOException {
+    if (this.interceptor != null) {
+      this.interceptor.intercept(request);
+    }
+    HttpHeaders headers = request.getHeaders();
+    String existing = (String) headers.get(GOOG_API_CLIENT);
```

Comment:
consider renaming this to googleApiClientHeader

Reply:
I guess it's in alignment with what is followed in this package.

Reply:
I think there is scope for improving the variable name since the current one is very generic. I will leave it up to you